File: Bachelor/5.semester/User Interface Programming/src/BooksService/BooksService.Common/Books/BookFilter.cs
namespace BooksService.Common
{
    public class BookFilter
    {
        public string Name { get; set; }
        public int? Year { get; set; }
        public string Isbn { get; set; }
        public string Author { get; set; }
        public string Genre { get; set; }
    }
}

File: Master/2.semester/Advanced Information Systems/src/Spring/src/main/java/com/fit/vut/Library/services/filters/BookFilter.java
package com.fit.vut.Library.services.filters;

import java.util.List;

public class BookFilter extends TitleFilter {

    private String isbn;

    private List<String> genres;

    public BookFilter(String name, String publisher, String language, String author, String isbn, List<String> genres){
        super(name, publisher, language, author);
        this.isbn = isbn;
        this.genres = genres;
    }

    public String getIsbn() {
        return isbn;
    }

    public void setIsbn(String isbn) {
        this.isbn = isbn;
    }

    public List<String> getGenres() {
        return genres;
    }

    public void setGenres(List<String> genres) {
        this.genres = genres;
    }
}
